part-1.py

```diff
@@ -4,17 +4,41 @@
 # the appropriate comment.
 
 # factorial
+def factorial(n):
 
+    if n == 0:
+        return 1
+
+    return n * factorial(n - 1)
 
 
 # reverse
+def reverse(text):
+
+    if len(text) <= 1:
+        return text
 
+    return text[-1] + reverse(text[:-1])
 
 
 # bunny
 
+def bunny(count):
 
+    if count == 0:
+        return 0
+    elif count == 1:
+        return 2
 
 # is_nested_parens
 
+def is_nested_parens(parens):
+
+    if parens == "":
+        return True
+    elif len(parens) % 2 != 0:
+        return False
```

Comment:
This works, but it's going to short circuit the recursion. A recursive base case is something more like:
```suggestion
    elif len(parens) == 1:
        return False
```